enhancements/etcd/disaster-recovery-with-ceo.md

```diff
@@ -19,86 +19,200 @@ see-also:
+restoring the corresponding key will make the etcd data unusable.
+    
+### Restoring the cluster state from backup 
+This solution handles situations where you want to restore your cluster to a previous state, for example, if something
+critical got deleted accidentally. This process is also used for recovering from a loss of quorum.
+    
+The script for restoring to a previous cluster state using a backup file will restore the etcd database from the backup
+along with the static pod resources for the entire control plane (etcd, kube-apiserver, kubecontrollermanager and
+kube-scheduler). See the implementation details for more information.
+    
+### Removal of a member from the etcd cluster
+In the past, we used a script to remove a member from the etcd cluster. In this release, the remove operation will be
+documented to invoke the `etcdctl member remove` command directly on the newly created `etcdctl` container.
+    
+### Recovering a single unhealthy member
+See the implementation details below.
+    
+### Recovering from a loss of quorum
+Recovering from a loss of quorum requires restoring from a previous backup. See the implementation details for
+`Restoring the cluster from backup`.
+   
+### Recovery of a member with a bad data-dir.
+If you have a majority of your masters still available and have an etcd quorum, then the following procedure should
+restore the data-dir:
+1. Stop etcd static pod by moving the etcd pod yaml out of kubelet manifest directory.
+1. Remove the bad data directory.
+1. Restart etcd static pod by moving the etcd pod yaml back into the kubelet manifest directory.
+
+If you have lost the majority of your master hosts, leading to etcd quorum loss, then, it requires restoring from a
+previous backup. See the section `Recovering from a loss of quorum.`
+
+###  All nodes being shut off at the same time and restarted.
+Fix the initialization process to detect lights-out scenario to restart the etcd without requiring any manual
+intervention.
+    
+### Upgrade, downgrade, re-upgrade
+Upgrade to 4.4, downgrade to 4.3, and re-upgrading to 4.4 should work, although downgrade is not officially supported.
 
 ## Implementation Details/Notes/Constraints
 
-*  Fix the version mismatch bug -- **done**.
-*  Change the name of the spec from `etcd-member.yaml` to `etcd-pod.yaml` -- **done**.
-*  Change the backup scripts to include the static pod resources of etcd -- **done**.
-*  Change the restore scripts to extract the static pod resources of etcd -- **done**.
-*  Fix the need to extract ETCD_DNS_NAME and  DISCOVERY_DOMAIN from pod spec -- **done**.
+### Restoring the cluster from backup
+In the prior releases, we restored the backup on all masters simultaneously for the cluster to come back up properly.
+With this release we implement a different approach of bringing a control plane up with a single node etcd cluster and
+then scaling it up to three.
+
+This new approach involves two steps:
```

Comment:
you need to start with something like, "establish ssh connectivity to *all* masters.  Part of this process will make the kube-apiserver inaccessible, so you will not be able to run `oc debug node`"